### ISSUE TITLE
Generic extended type isn't taken into account

### ISSUE THREAD

**Issue by RemyMachado** [microsoft/TypeScript#46131]
# Bug Report

### 🔎 Search Terms
Extend, type, error, Property does not exist on type, TS2339, generic

### ⏯ Playground Link
https://snack.expo.dev/ZWoT7swlU

### 💻 Code
```typescript
import React from 'react'
import { NativeScrollEvent, NativeSyntheticEvent, ScrollView } from 'react-native'

const CreateRestitutionInformation = () => {
  const isCloseToBottom = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
    const paddingToBottom = 20
    
    // Error next line
    const result = event.layoutMeasurement.height + event.contentOffset.y >= event.contentSize.height - paddingToBottom
    ...
  }

  return (
    <ScrollView onMomentumScrollEnd={isCloseToBottom}>
      ...
    </ScrollView>
  )
}

export default CreateRestitutionInformation
```

### 🙁 Actual behavior
TypeScript gives the following error: `TS2339: Property 'layoutMeasurement' does not exist on type 'NativeSyntheticEvent '`.
It doesn't recognize any of the three properties I'm trying to use (`event.layoutMeasurement`, `event.contentOffset`, `event.contentSize`).

### 🙂 Expected behavior
TypeScript isn't expected to throw an error.

Here's the definitions that are imported:
```typescript
export interface NativeSyntheticEvent<T> extends React.BaseSyntheticEvent<T, NodeHandle, NodeHandle> {}
```
```typescript
export interface NativeScrollEvent {
    contentInset: NativeScrollRectangle;
    contentOffset: NativeScrollPoint;
    contentSize: NativeScrollSize;
    layoutMeasurement: NativeScrollSize;
    velocity?: NativeScrollVelocity | undefined;
    zoomScale: number;
    /**
     * @platform ios
     */
    targetContentOffset?: NativeScrollPoint | undefined;
}
```

I think I'm missing something, it's such a weird behavior 🤔

EDIT:
I was not using the event properly.
It should be: `event.nativeEvent.layoutMeasurement.height` instead of `event.layoutMeasurement.height`

**Comment by ksielyov** [comment]
Same
